c2rust-transpile/src/translator/assembly.rs

```diff
@@ -722,10 +715,9 @@ impl<'c> Translation<'c> {
                 if let Ok(idx) = ref_str.parse::<usize>() {
                     outputs
                         .iter()
-                        .chain(inputs.iter())
+                        .chain(inputs)
                         .nth(idx)
-                        .map(operand_is_mem_only)
-                        .unwrap_or(false)
+                        .map_or(false, operand_is_mem_only)
                 } else {
                     false
                 }
```

Comment:
This could be rewritten as:

```rust
ref_str.parse::<usize>().ok().and_then(|idx| {
    outputs
        .iter()
        .chain(inputs)
        .nth(idx)
        .map(operand_is_mem_only)
}).unwrap_or_default()
```